src/operator/tensor/init_op.h

```diff
@@ -471,6 +473,11 @@ inline bool RangeShape(const nnvm::NodeAttrs& attrs,
     << "Range does not support step=0, received " << param.step;
   CHECK(param.repeat > 0)
     << "Range only supports repeat > 0, received " << param.repeat;
+  if (param.start == param.stop.value()) {
```

Comment:
though I understand that end=None is already taken, this condition still feels a bit like a hack...

Reply:
@szha what's the best way to proceed? it seems that everyone is very busy, so doing a discussion over PR comments might take a very long time. ideal would be some kind of realtime chat where we can discuss design alternatives and quickly come to a consensus. is there such a forum?

Reply:
if that is not desired, here's a simple proposal: introduce a new parameter called "infer_range" that defaults to false. if it is true, then exactly one of `param.stop`, `param.start`, or `param.step` can be None, and will be inferred from the others and the output dimensions. I may only implement something more limited that makes `param.stop` be the only inferrable parameter, and leave that to others to implement in the future.

Reply:
Adding a flag sounds good to me